analyses/new-feature-template.qmd

```diff
@@ -0,0 +1,1549 @@
+
+## Neighborhood Values
+
+::: panel-tabset
+
+### Mean feature
+
+```{r mean_feature}
+if (params$type == "categorical") {
+  spatial_data %>%
+    ggplot() +
+    geom_sf(aes(fill = !!sym(paste0(params$added_feature, "_plurality_type")))) +
+    scale_fill_viridis_d(option = "viridis", name = "Type of Plurality") +
+    geom_sf(aes(alpha = !!sym(paste0(params$added_feature, "_plurality_percentage")))) +
+    scale_alpha_continuous(name = "Percentage of Plurality") +
+    theme_void() +
+    coord_sf(xlim = c(-88.4, -87.52398), ylim = c(41.5, 42.2))
+} else {
+  spatial_data %>%
+    ggplot() +
+    geom_sf(aes(fill = !!sym(paste0(params$added_feature, "_neighborhood_mean")))) +
+    scale_fill_viridis_c(option = "viridis", name = "Value") +
+    theme_void() +
+    coord_sf(xlim = c(-88.4, -87.52398), ylim = c(41.5, 42.2))
+}
+```
+
+### Mean Absolute SHAP
+
+```{r mean_shap}
+spatial_data %>%
+  ggplot() +
+  geom_sf(aes(fill = !!sym(paste0(params$added_feature_shap, "_neighborhood_mean")))) +
+  scale_fill_viridis_c(option = "viridis", name = "Value") +
+  theme_void() +
+  coord_sf(xlim = c(-88.4, -87.52398), ylim = c(41.5, 42.2))
+```
+
+### 90th Percentile of Absolute SHAP
+
+```{r 90th_percentile_shap}
+spatial_data %>%
+  ggplot() +
+  geom_sf(aes(fill = !!sym(paste0(params$added_feature_shap, "_neighborhood_90th")))) +
+  scale_fill_viridis_c(option = "viridis", name = "Value") +
+  theme_void() +
+  coord_sf(xlim = c(-88.4, -87.52398), ylim = c(41.5, 42.2))
+```
+
+### Median change in Neighborhood FMV
+This value is defined as the neighborhood level median increase in value when adding the new feature to the model. For example, a value of 1% would mean that adding the feature increased properties within a neighborhood by 1%.
+
+```{r neighborhood_change}
+assessment_pin %>%
+  select(meta_pin, loc_longitude, loc_latitude, pred_pin_final_fmv) %>%
+  rename(pred_pin_final_fmv_new = pred_pin_final_fmv) %>%
+  inner_join(assessment_pin_comparison, by = "meta_pin") %>%
+  rename(pred_pin_final_fmv_comparison = pred_pin_final_fmv) %>%
+  group_by(meta_nbhd_code) %>%
+  summarize(
+    median_fmv_new = median(pred_pin_final_fmv_new, na.rm = TRUE),
+    median_fmv_comparison = median(pred_pin_final_fmv_comparison, na.rm = TRUE),
+    fmv_ratio = (median_fmv_new / median_fmv_comparison) / median_fmv_comparison
+  ) %>%
+  inner_join(nbhd, by = c("meta_nbhd_code" = "town_nbhd")) %>%
+  st_as_sf() %>%
+  ggplot() +
+  geom_sf(aes(fill = fmv_ratio)) +
+  scale_fill_viridis_c(option = "viridis", name = "FMV Ratio", labels = scales::percent_format(accuracy = 0.001)) +
+  theme_void() +
+  coord_sf(xlim = c(-88.4, -87.52398), ylim = c(41.5, 42.2))
+```
+
+:::
+
+
+```{r leaflet_function}
+create_leaflet_map <- function(dataset, legend_value, legend_title, order_scheme = "high", longitude = "loc_longitude", latitude = "loc_latitude") {
+  # Filter neighborhoods that have at least one observation
```

Comment:
Do we want an include /exclude based on neighborhood borders?

Reply:
Not sure what this means. Can you clarify?

Reply:
Do we want all maps to have neighborhood borders? Some maps show all neighborhoods, others only show a few.

Reply:
If we're showing all neighborhoods then we probably don't want borders, if it's just a couple then I find borders useful.